### PR TITLE
Add config for objectId size

### PR DESCRIPTION
**Pull request by steven-supersolid**
The current objectId length of 10 mixed case alphanumeric characters is only really suitable for classes that contain up to 1 million objects.

This review adds a new constructor option objectIdSize which defaults to the current value of 10, can also be set via the environment variable PARSE_OBJECT_ID_SIZE.

### CODE REVIEW

**Comment by TylerBrock** [comment]
We might want to add a test to ensure that even when increasing/decreasing the size of an objectId values that are shorter/longer than the current setting still are able to be read and processed without any errors.

**Comment by flovilmart** [comment]
Let’s make it for 2.6.0, i’d Like to release 2.5.0 ASAP :)

**Comment by steven-supersolid** [comment]
Added a test that creates an object with a short objectId then increases the objectId size setting to perform a find, then an update.

The second part of the test creates an object with a long objectId and decreases the objectId size setting to perform a find only. Let me know if you don't think this is sufficient, or overkill as not sure part two is required - increasing the setting should be no different to decreasing, as long as there is no code that checks inequalities. I don't think we actually check anything with the objectId length but could be wrong.

File: src/cryptoUtils.js [hunk 1]
```diff
@@ -35,9 +35,8 @@ export function randomString(size: number): string {
 }
 
 // Returns a new random alphanumeric string suitable for object ID.
-export function newObjectId(): string {
-  //TODO: increase length to better protect against collisions.
```
Comment (natanrolnik) [inline src/cryptoUtils.js]: Finally someone took care of this `//TODO` 😄

**Comment by natanrolnik** [comment]
Waiting release of 2.5.0 to merge this

File: src/cryptoUtils.js (hunk 1 above)
Comment (steven-supersolid) [inline src/cryptoUtils.js]: It's funny, I must have seen that comment over a year ago and it never registered that there might be a problem :)

**Comment by Moumouls** [comment]
Hi @steven-supersolid ,
I set up the environment variables "PARSE_OBJECT_ID_SIZE" but apparently this does not change the size of the ID (I have moved to 20). I'm in version 2.6 of Parse Server. Use the environment variables is the good way ?
capture d ecran 2017-10-10 a 15 48 33

Thanks for this add :)
(Sorry for the English, i'm french ) ;)

**Comment by steven-supersolid** [comment]
Using the environment variable should work if you launch from the CLI but won't work when mounted as express middleware.

The setting will also not affect any old id's and I'm unsure how the id's will display in the dashboard, so recommend checking your database entries too.

**Comment by Moumouls** [comment]
@steven-supersolid Ok, my Parse server is mounted with express, I will pass the objectIdSize parameter in the constructor, thanks for the quick response! It Works !
capture d ecran 2017-10-10 a 16 28 36
capture d ecran 2017-10-10 a 16 28 06

**Pull request merged**